src/memote/reactions.jl

```diff
@@ -4,3 +4,115 @@ M.E., Olivier, B.G. et al. MEMOTE for standardized genome-scale metabolic model
 testing. Nat Biotechnol 38, 272–276 (2020).
 https://doi.org/10.1038/s41587-020-0446-y for details.
 =#
+
+"""
+$(TYPEDSIGNATURES)
+
+Test if a reaction is constrained.
+"""
+_is_constrained(lb, ub, default) = lb ∉ [-default, 0, default] || ub ∉ [-default, 0, default] 
+
+"""
+$(TYPEDSIGNATURES)
+
+Find all purely metabolic reactions and indicate which are constrained beyond
+directionality (if value of dictionary is true, then constrained). For a
+constraint to be not purely directional, the lower or upper bound needs to be
+different from `[-config.reactions.bound_default, 0,
+config.reactions.bound_default]`. Metabolic reactions exclude transport,
+boundary and biomass reactions.
+"""
+function find_all_purely_metabolic_reactions(model; config = memote_config)
+    biomass_rxns = model_biomass_reactions(model; config)
+    metabolic_reactions = Dict{String, Bool}()
+    for (lb, ub, rid) in zip(bounds(model)..., reactions(model))
+        is_boundary(model, rid) && continue
+        rid in biomass_rxns && continue
+        _probably_transport_reaction(model, rid, config.reaction.test_annotation) && continue
+        metabolic_reactions[rid] = _is_constrained(lb, ub, config.reaction.bound_default)
+    end
+    return metabolic_reactions
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Find all transport reactions and indicate which are constrained beyond
+directionality. For a constraint to be not purely directional, the lower or
+upper bound needs to be different from `[-config.reactions.bound_default, 0,
+config.reactions.bound_default]`. Transport reactions are heuristically
+identified, see [`!_probably_transport_reaction`](@ref), which partially uses
+reaction annotations. Set the annotation field to use via
+`config.reactions.rest_annotation`. 
+"""
+function find_all_transport_reactions(model; config = memote_config)
+    transport_reactions = Dict{String, Bool}()
+    for (lb, ub, rid) in zip(bounds(model)..., reactions(model))
+        !_probably_transport_reaction(model, rid, config.reaction.test_annotation) && continue
+        transport_reactions[rid] = _is_constrained(lb, ub, config.reaction.bound_default)
+    end
+    return transport_reactions
+end
+
+"""
+$(TYPEDSIGNATURES)
+
+Find all reactions with overlapping annotation information. Internally calls
+[`annotation_index`](@ref). Some annotations, like sbo terms will necessarily be
+non-unique, ignore annotations like this by editing `config.reaction.ignore_annotations`.
+"""
+function reactions_with_partially_identical_annotations(model; config = memote_config)
+    stdmodel = convert(StandardModel, model)
+    idx = annotation_index(stdmodel.reactions)
+    for anno in config.reaction.ignore_annotations
+        delete!(idx, anno)
+    end
```

Comment:
this can be done much easier as a `filter`

Reply:
why is the current way not easy?